### PR TITLE
Add circleci support for test & deploy

### LINKED ISSUE #99: Add circleci support
@dunglas WDYT about adding circleci support?

### PR DESCRIPTION
**Pull request by vincentchalamon**
Fix #99

- [x] Configure circleci
- [x] Run jobs in parallel
- [x] Add deploy through circleci, if configuration is correct
- [x] Add test tools: phpunit, phpstan, php-cs-fixer, monitor health, swagger validator
- [x] Update symfony/* dependencies to 4.2.8 (security fix)
- [x] Remove travis support
- [x] Configure deploy env var in CircleCI (ping @dunglas)
- [x] Check deploy is correct

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -0,0 +1,329 @@
+# PHP CircleCI 2.0 configuration file
```
Comment (teohhanhui) [inline .circleci/config.yml]: Could we remove the header?

File: admin/Dockerfile [hunk 2]
```diff
@@ -4,16 +4,14 @@ RUN mkdir -p /usr/src/admin
...
-
 RUN apk add --no-cache --virtual .gyp \
         python \
         make \
-        g++ \
-	&& yarn install \
-	&& apk del .gyp
+        g++
 
+# Prevent the reinstallation of node modules at every changes in the source code
```
Comment (teohhanhui) [inline admin/Dockerfile]: The comment contradicts what's happening here.

**Comment by vincentchalamon** [comment]
@api-platform/core-team Travis kills its script without any error nor output...

**Comment by teohhanhui** [comment]
> Travis kills its script without any error nor output...

Seems like a problem with installing the Blackfire extension.

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+              apt install nodejs
+              source ci/before_deploy
+              ci/deploy
+            else
+              echo "Deployment skipped"
+            fi
+
+workflows:
+  version: 2
+  test-and-deploy:
```
Comment (teohhanhui) [inline .circleci/config.yml]: Uhh, I've never realized I was doing the workflows incorrectly in api-platform/core :laughing:

I'll go fix that.

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+    jobs:
+      # API
+      - init-api
+      - security_checker:
```
Comment (teohhanhui) [inline .circleci/config.yml]: Can we be consistent and use dashes for all the job names? :smile:

File: api/.php_cs.dist [hunk 5]
```diff
@@ -0,0 +1,37 @@
+<?php
+
+$finder = PhpCsFixer\Finder::create()
+    ->exclude(['bin', 'config', 'docker', 'helm', 'var', 'vendor'])
```
Comment (teohhanhui) [inline api/.php_cs.dist]: Only `var` is necessary here, I think.

`bin` and `config` should not be excluded.

`vendor` is already excluded by default: https://github.com/FriendsOfPHP/PHP-CS-Fixer/blob/v2.15.0/src/Finder.php#L33

The rest don't contain *.php files, I suppose?

File: api/.php_cs.dist [hunk 6]
```diff
@@ -0,0 +1,37 @@
...
+    ->in(__DIR__)
+;
+
+return PhpCsFixer\Config::create()
+    ->setRiskyAllowed(true)
+    ->setRules([
+        '@DoctrineAnnotation' => true,
+        '@Symfony' => true,
+        'array_syntax' => ['syntax' => 'short'],
```
Comment (teohhanhui) [inline api/.php_cs.dist]: Better to keep the nesting level consistent by expanding the array, even if it's only one option for now.

File: .circleci/config.yml (hunk 3 above)
Comment (vincentchalamon) [reply .circleci/config.yml]: I just checked your code. Didn't know orbs were available for free accounts, will try to implement them in this configuration

**Comment by dunglas** [comment]
Can you rebase please?

**Pull request merged**